### PR TITLE
script to get modified objects and index them

### PR DESCRIPTION
**Pull request by artntek**
Script for use when moving legacy installations to k8s. Gets objects modified since a certain date, and indexes them

see #1990

### CODE REVIEW

File: src/scripts/bash/k8s/index-delta.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    echo "Response:"
+    curl_result=$(curl  -s  -X  PUT  --cert ${cert_file}  "${api_url}?pid=${pid}")
+    echo "${curl_result}"
+    echo
+    if [ "$(echo "${curl_result}" | grep -cz "<scheduled>\s*true\s*</scheduled>")" -lt 1 ]; then
+        echo
+        echo "Error: indexing failed -- exiting..."
+        echo
+        cleanup
+        exit 2
```
Comment (taojing2002) [inline src/scripts/bash/k8s/index-delta.sh:92]: Is it possible to write the failed pid into a file and the index still keeps doing the job? Later we can investigate the failed pids.  So one failure will not halt the whole process.

File: src/scripts/bash/k8s/index-delta.sh [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+# Now, index the objects
+result_count=$(cat ${result_file} | wc -w | xargs)
+echo "Found ${result_count} objects created/modified since ${start_time}, in release"
+echo "${release_name}."
+echo
+echo "Calling ${api_url} to index each. Enter to continue or Ctrl+C to abort..."
+read
+
+idx=0
+for pid in $(cat ${result_file}); do
```
Comment (taojing2002) [inline src/scripts/bash/k8s/index-delta.sh:79]: Is it possible to put two or thee pids into one curl command so we can cut some network traffics? It depends on how many pids need to be indexed. If it is just dozens, maybe it is not worth to do it.

Comment (mbjones) [reply src/scripts/bash/k8s/index-delta.sh:79]: If there are a lot of PIDs, `parallel` or `xargs -j` would likely be much faster than the serial loop. See: https://learning.nceas.ucsb.edu/2024-03-arctic/sections/parallel-programming.html#parallel-processing-in-the-shell

Comment (artntek) [reply src/scripts/bash/k8s/index-delta.sh]: Thanks. The requests are very fast (they just queue the pids for indexing then return), and I don't foresee there being enough pids to warrant parallelizing (eg for ADC, there were just over 3k pids to reindex in the last month).

Comment (artntek) [reply src/scripts/bash/k8s/index-delta.sh]: batched in groups of 5 pids per call

File: src/scripts/bash/k8s/index-delta.sh (hunk 1 above)
Comment (artntek) [reply src/scripts/bash/k8s/index-delta.sh]: made this change

**Review by taojing2002 — APPROVED** [review]

**Pull request merged**